src/app/header/header.component.html

```diff
@@ -1,76 +1,157 @@
+        </a>
+        <ul class="dropdown-menu">
+          <li *ngFor="let m of dummy.contexts">
+            <a [routerLink]="[urlFeatureToggle + m.path]" *ngIf="m.path !== null && m != context.current">
+              <span class="nav-item-icon">
+                <span class="nav-icon {{m.type.icon}}"></span>
+              </span>
+              <span class="nav-item-text">{{m.name}}</span>
+            </a>
+            <a *ngIf="m.path === null">
+              <span class="nav-item-icon">
+                <span class="nav-icon {{m.type.icon}}"></span>
+              </span>
+              <span class="nav-item-text">{{m.name}}</span>
+            </a>
+          </li>
+          <li>
+            <a [routerLink]="[urlFeatureToggle + '/pmuir/spaces']">
+              <span class="nav-item-icon">
+              </span>
+              <span class="nav-item-text">
+                Browse all
+              </span>
+            </a>
+          </li>
+          <li class="divider"></li>
+          <li>
+            <a [routerLink]="[urlFeatureToggle + '/home']">
+              <span class="nav-item-icon">
+              </span>
+              <span class="nav-item-text">
+                Account home
+              </span>
+            </a>
+          </li>
+          <li>
+            <!-- TODO: This should pop up the new space dialog -->
+            <a>
+              <span class="nav-item-icon"></span>
+              <span class="nav-item-text">
+                New space
+              </span>
+            </a>
+          </li>
         </ul>
-    </div>
+      </li>
+      <li *ngFor="let m of context.current.type.menus" [class.active]="m.active" [class.with-no-children]="(m.menus||[]).length===0">
+        <a [routerLink]="[urlFeatureToggle + m.fullPath]">
+          <span *ngIf="m.icon" class="nav-item-icon">
+            <span  class="nav-icon {{m.icon}}"></span>
+          </span>
+          <span *ngIf="m.name" class="nav-item-text">{{m.name}}</span>
+        </a>
+        <ul class="nav navbar-nav navbar-persistent" *ngIf="m.menus">
+          <li *ngFor="let n of m.menus" [class.active]="n.active">
+            <a [routerLink]="[urlFeatureToggle + n.fullPath]">
+              <span class="nav-item-text">
+                {{n.name}}
+              </span>
+            </a>
+          </li>
+        </ul>
+      </li>
+    </ul>
+    <ul id="header_rightDropdown" class="nav navbar-nav navbar-right">
+      <li>
+        <a *ngIf="!loggedIn" (click)='login();'>Sign In</a>
+      </li>
+      <li *ngIf="loggedIn" dropdown class="pull-right dropdown">
+        <a dropdown-open>
+          <div *ngIf="loggedInUser" id="header_dropdownToggle2">
+            <span class="nav-item-icon">
+              <span *ngIf="!imgLoaded" class="nav-icon pficon-user"></span>
+              <img (load)="onImgLoad()" src="{{ loggedInUser.attributes?.imageURL }}" [ngClass]="{'show' : imgLoaded,'img-icon' : imgLoaded, 'hide' : !imgLoaded}">
```

Comment:
i think it's a caching issue and unrelated to menu visual appearance which is the purpose of this pull request ... I suspect this is an issue with the login code